src/Microsoft.AspNetCore.Routing/Template/TemplatePart.cs

```diff
@@ -5,12 +5,47 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
+using Microsoft.AspNetCore.Routing.Patterns;
 
 namespace Microsoft.AspNetCore.Routing.Template
 {
     [DebuggerDisplay("{DebuggerToString()}")]
     public class TemplatePart
     {
+        public TemplatePart()
+        {
+        }
+
+        public TemplatePart(RoutePatternPart other)
+        {
+            IsLiteral = other.IsLiteral || other.IsSeparator;
+            IsParameter = other.IsParameter;
+
+            if (other.IsLiteral && other is RoutePatternLiteralPart literal)
+            {
+                Text = literal.Content;
+            }
+            else if (other.IsParameter && other is RoutePatternParameterPart parameter)
+            {
+                // Text is unused by TemplatePart and assumed to be null when the part is a parameter.
+                Name = parameter.Name;
+                IsCatchAll = parameter.IsCatchAll;
+                IsOptional = parameter.IsOptional;
+                DefaultValue = parameter.Default;
+                InlineConstraints = parameter.Constraints?.Select(p => new InlineConstraint(p));
+            }
+            else if (other.IsSeparator && other is RoutePatternSeparatorPart separator)
+            {
+                Text = separator.Content;
+                IsOptionalSeperator = true;
+            }
+            else
+            {
+                // Unreachable
+                throw new NotSupportedException();
```

Comment:
Is this an ok thing to do or was it done in a hurry?

Reply:
It should be unreachable code. What would you suggest?

Reply:
If you're fine with it, I'm fine with it. Just checking